app/api/ada/adaTransactions/adaNewTransactions.test.js

```diff
@@ -0,0 +1,204 @@
+
+// @flow
+import BigNumber from 'bignumber.js';
+import '../lib/test-config';
+import type { UTXO, AdaAddress } from '../adaTypes';
+import {
+  newAdaUnsignedTx,
+  newAdaUnsignedTxFromUtxo,
+  sendAllUnsignedTxFromUtxo,
+  signTransaction
+} from './adaNewTransactions';
+import {
+  NotEnoughMoneyToSendError,
+} from '../errors';
+
+import { RustModule } from '../lib/cardanoCrypto/rustLoader';
+
+// function to mock our network call
+function makeNetworkMock(utxos) {
+  return async (addresses) => {
+    const senderUtxos = [];
+    for (const addr of addresses) {
+      for (const utxo of utxos) {
+        if (utxo.receiver === addr) {
+          senderUtxos.push(utxo);
+        }
+      }
+    }
+    return senderUtxos;
+  };
+}
+
+const sampleUtxos: Array<UTXO> = [
+  {
+    amount: '7000',
+    receiver: 'Ae2tdPwUPEZKX8N2TjzBXLy5qrecnQUniTd2yxE8mWyrh2djNpUkbAtXtP4',
+    tx_hash: '05ec4a4a7f4645fa66886cef2e34706907a3a7f9d88e0d48b313ad2cdf76fb5f',
+    tx_index: 0,
+    utxo_id: '05ec4a4a7f4645fa66886cef2e34706907a3a7f9d88e0d48b313ad2cdf76fb5f0',
+  },
+  {
+    amount: '1000000',
+    receiver: 'Ae2tdPwUPEZKX8N2TjzBXLy5qrecnQUniTd2yxE8mWyrh2djNpUkbAtXtP4',
+    tx_hash: '6930f123df83e4178b0324ae617b2028c0b38c6ff4660583a2abf1f7b08195fe',
+    tx_index: 0,
+    utxo_id: '6930f123df83e4178b0324ae617b2028c0b38c6ff4660583a2abf1f7b08195fe0',
+  },
+  {
+    amount: '10000000',
+    receiver: 'Ae2tdPwUPEZ4xAL3nxLq4Py7BfS1D2tJ3u2rxZGnrAXC8TNkWhTaz41J3FN',
+    tx_hash: '0df0273e382739f8b4ae3783d81168093e78e0b48ec2c5430ff03d444806a173',
+    tx_index: 0,
+    utxo_id: '0df0273e382739f8b4ae3783d81168093e78e0b48ec2c5430ff03d444806a1730',
+  },
+];
+
+const sampleAdaAddresses: Array<AdaAddress> = [
+  {
+    cadAmount: { getCCoin: new BigNumber(0) },
+    cadId: 'Ae2tdPwUPEZEtwz7LKtJn9ub8y7ireuj3sq2yUCZ57ccj6ZkJKn7xEiApV9',
+    cadIsUsed: false,
+    account: 0,
+    change: 1,
+    index: 11,
+  },
+  {
+    cadAmount: { getCCoin: new BigNumber(0) },
+    cadId: 'Ae2tdPwUPEZKX8N2TjzBXLy5qrecnQUniTd2yxE8mWyrh2djNpUkbAtXtP4',
+    cadIsUsed: false,
+    account: 0,
+    change: 0,
+    index: 135,
+  },
+];
+
+beforeAll(async () => {
+  await RustModule.load();
+});
+
+describe('Create unsigned TX from UTXO', () => {
+  it('Should create a valid transaction withhout selection', async () => {
+    const utxos: Array<UTXO> = [sampleUtxos[1]];
+    const unsignedTxResponse = await newAdaUnsignedTxFromUtxo(
+      'Ae2tdPwUPEZKX8N2TjzBXLy5qrecnQUniTd2yxE8mWyrh2djNpUkbAtXtP4',
+      '5000', // smaller than input
+      null,
+      utxos
+    );
+    expect(unsignedTxResponse.senderUtxos).toEqual(utxos);
+    expect(unsignedTxResponse.txBuilder.get_input_total().to_str()).toEqual('1.000000');
+    expect(unsignedTxResponse.txBuilder.get_output_total().to_str()).toEqual('0.005000');
+    expect(unsignedTxResponse.txBuilder.estimate_fee(
```

Comment:
I’m confused with his one. Should we be testing `estimate_fee` if we are not using it and we are actually using `get_balance_without_fees`?

Reply:
I didn't see a reason not to add it. Didn't give it much thought.
It could change (without affecting `get_balance_without_fees`) if we change the selection policy